{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8187d3ec-eba7-4253-af5e-079963ee7b27",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d009ce92-177b-4c74-af09-d041693204bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Token</th>\n",
       "      <th>Tool Name?</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADAP-GC</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AMDIS</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AnalyzerPro</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ArMet</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ArrayTrack</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Token Tool Name?\n",
       "0      ADAP-GC        Yes\n",
       "1        AMDIS        Yes\n",
       "2  AnalyzerPro        Yes\n",
       "3        ArMet        Yes\n",
       "4   ArrayTrack        Yes"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read wlads manual curations\n",
    "wlad_cur = pd.read_csv(\"Data/150_Bbatch_query1.csv\")\n",
    "wlad_cur2 = pd.read_csv(\"Data/300_Abatch_query2only.csv\")\n",
    "wlad_cur3 = pd.read_csv(\"Data/batchb_resample_query1_2020-06-16.csv\")\n",
    "wlad_cur3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7ed5805c-ae48-4313-84cb-e78445ded4fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get tools into list\n",
    "#wlad_tools = wlad_cur.token.values.tolist() + wlad_cur2.Token.values.tolist()\n",
    "wlad_tools = wlad_cur3.Token.values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "03630a44-8d7b-439c-8b15-a9650ba0e41d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>title_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29718355</td>\n",
       "      <td>MetExplore: collaborative edition and explorat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>21523460</td>\n",
       "      <td>A survey of metabolic databases emphasizing th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>33777285</td>\n",
       "      <td>A framework for stability-based module detecti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24165880</td>\n",
       "      <td>EBI metagenomics--a new resource for the analy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>31035005</td>\n",
       "      <td>Computational solutions in redox lipidomics - ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       pmid                                     title_abstract\n",
       "0  29718355  MetExplore: collaborative edition and explorat...\n",
       "1  21523460  A survey of metabolic databases emphasizing th...\n",
       "2  33777285  A framework for stability-based module detecti...\n",
       "3  24165880  EBI metagenomics--a new resource for the analy...\n",
       "4  31035005  Computational solutions in redox lipidomics - ..."
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trainCurate = pd.read_csv(\"Data/manualCurate.csv\")\n",
    "trainCurate2 = pd.read_csv(\"Data/query2_manualCurate.csv\")\n",
    "trainCurate1_2 = pd.read_csv(\"Data/query1_2-manualCurate.csv\")\n",
    "trainCurate1_2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2140fe16-5f68-4450-ad36-f569bfeaef06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Metabolomics: Challenges and Opportunities in Systems Biology Studies. Metabolomics has the capability of providing predisposition, diagnostic, prognostic, and therapeutic biomarker profiles of individual patients, since a large number of metabolites can be measured in an unbiased manner from biological samples. In this setting, <sup>1</sup>H-Nuclear Magnetic Resonance (NMR) spectroscopy of biofluids such as plasma, urine, and fecal water offers the opportunity to identify patterns of biomarker changes that reflects the physiological or pathological status of an individual patient.In this chapter, we show as a metabolomics study can be used to diagnose a disease, classifying patients as healthy or as pathological taking into account individual variability.'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trainCurate1_2.title_abstract[300]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b8d9cda2-8009-495b-a904-22a2efa1fb7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# From Query 1.2\n",
    "toolsFound = set([\n",
    "    'MetExplore',\n",
    "    'MetaCyc',\n",
    "    'KEGG',\n",
    "    'Reactome',\n",
    "    'BiGG',\n",
    "    'bootcluster',\n",
    "    'EBI',\n",
    "    'Weka',\n",
    "    'PiMP',\n",
    "    'MetNet3D',\n",
    "    'MetNetDB',\n",
    "    'MeltDB',\n",
    "    'HAMMER',\n",
    "    'Mass Frontier',\n",
    "    'Dolphin',\n",
    "    'ProbMetab',\n",
    "    'isoMETLIN',\n",
    "    'MZmine',\n",
    "    'ProteoWizard',\n",
    "    'OmicsTract',\n",
    "    'SynpExtractor',\n",
    "    'MetaboLights',\n",
    "    'Dr.DMASS',\n",
    "    'KNApSAcK',\n",
    "    'BioSM',\n",
    "    'Xtrack',\n",
    "    'Sesame',\n",
    "    'HalX',\n",
    "    'SMfinder',\n",
    "    'LipidIMMS',\n",
    "    'ADAP',\n",
    "    'MESSI',\n",
    "    'PIPINO',\n",
    "    'CANTARE',\n",
    "    'specmine',\n",
    "    'ADAP-GC',\n",
    "    'OmicsVis',\n",
    "    'MSFACTs',\n",
    "    'MetaNetter',\n",
    "    'IPAD',\n",
    "    'ProMetab',\n",
    "    'Qualis-SIS',\n",
    "    'ATTED-II',\n",
    "    'SIEVE',\n",
    "    'XCMS Online',\n",
    "    'Compound Discoverer',\n",
    "    'metaP-server',\n",
    "    'AStream',\n",
    "    'Omix',\n",
    "    'HiRes',\n",
    "    'MetaboSearch',\n",
    "    'MetaboAnalyst',\n",
    "    'MultiOmics Explainer',\n",
    "    'BioContainers',\n",
    "    'iSCHRUNK',\n",
    "    'PCAtoTree',\n",
    "    'ChemDistiller',\n",
    "    'MS-CleanR',\n",
    "    'MS-FINDER',\n",
    "    'MS-DIAL',\n",
    "    'MPA-RF',\n",
    "    'TOPPView',\n",
    "    'EIder',\n",
    "    'RDFIO',\n",
    "    '%polynova_2way',\n",
    "    'LiverWiki',\n",
    "    'LipidXte',\n",
    "    'EasyLCMS',\n",
    "    'MetNet',\n",
    "    'ArMet',\n",
    "    'MZmine',\n",
    "    'CAMERA',\n",
    "    'credentialing',\n",
    "    'Warpgroup',\n",
    "    'metaXCMS',\n",
    "    'X(13)CMS',\n",
    "    'AMON',\n",
    "    'GOcats',\n",
    "    'Map2Slim',\n",
    "    'NOREVA',\n",
    "    'Metab',\n",
    "    'AMDIS',\n",
    "    'Massifquant',\n",
    "    'centWave',\n",
    "    'matchedFilter',\n",
    "    'ARACNE',\n",
    "    'LUCIDus',\n",
    "    'KnowEnG',\n",
    "    'EigenMS',\n",
    "    'jmzTab',\n",
    "    'Automics',\n",
    "    'DECOMP',\n",
    "    'KIMBLE',\n",
    "    'MUSCLE',\n",
    "    'MetPP',\n",
    "    'ECOMICS',\n",
    "    'TICL',\n",
    "    'MetPA',\n",
    "    'pyTFA',\n",
    "    'matTFA',\n",
    "    'SMART',\n",
    "    'LocFuse',\n",
    "    'IIS',\n",
    "    'MoDentify',\n",
    "    'NExT',\n",
    "    'OutCyte',\n",
    "    'PhenoMeNal',\n",
    "    'BioMagResBank',\n",
    "    'mMass',\n",
    "    'IsoCor',\n",
    "    'MeMo',\n",
    "    'Heatmapper',\n",
    "    'COBRApy',\n",
    "    'IPA',\n",
    "    'IsoMS-Quant',\n",
    "    'CircaInSilico',\n",
    "    'SIMBioMS',\n",
    "    'BiNChE',\n",
    "    'OSCA',\n",
    "    'Maltcms',\n",
    "    'ConsensusPathDB',\n",
    "    'OpenMS',\n",
    "    'MarVis-Filter',\n",
    "    'mzMatch',\n",
    "    'IDEOM',\n",
    "    'iReMet-Flux',\n",
    "    'FluxFix',\n",
    "    'MWASTools',\n",
    "    'RankProd',\n",
    "    'Greazy',\n",
    "    'ComPath',\n",
    "    'METLIN',\n",
    "    'isoMETLIN',\n",
    "    'METLIN-MRM',\n",
    "    'STATegra',\n",
    "    'DrugBank',\n",
    "    'mSpecs',\n",
    "    'MetAssimulo',\n",
    "    'Chenomx',\n",
    "    'SASMeQ',\n",
    "    'SysTox-OM',\n",
    "    'CEBS',\n",
    "    'rNMR',\n",
    "    'MarkerView',\n",
    "    'BioSMXpress',\n",
    "    'BioSM',\n",
    "    'R.JIVE',\n",
    "    'MetaboMiner',\n",
    "    'IntLIM',\n",
    "    'Risa',\n",
    "    'jQMM',\n",
    "    'Metabox',\n",
    "    'DoGEX',\n",
    "    'Markerlynx',\n",
    "    'MetAlign',\n",
    "    'ROMANCE',\n",
    "    'LIMSA',\n",
    "    'SECD',\n",
    "    'YMDB',\n",
    "    'Omics Viewer',\n",
    "    'MBROLE',\n",
    "    'MBROLE2',\n",
    "    'MetaboGroup S',\n",
    "    'Pathview',\n",
    "    'Pathview Web',\n",
    "    'CcpNmr',\n",
    "    'GINsim',\n",
    "    'BioLQM',\n",
    "    'Pint',\n",
    "    'MaBoSS',\n",
    "    'Cell Collective',\n",
    "    'Haystack',\n",
    "    'SoyKB',\n",
    "    'KNApSAcK-3D',\n",
    "    'KNApSAcK',\n",
    "    'CompExtractor',\n",
    "    'MetaboLab',\n",
    "    'NMRLab',\n",
    "    'Guineu',\n",
    "    'MetNet',\n",
    "    'PathFinder',\n",
    "    'PathVisio',\n",
    "    'Workflow4Metabolomics',\n",
    "    'iMet-Q',\n",
    "    'MetAlign',\n",
    "    'Metscape',\n",
    "    'SIRIUS',\n",
    "    'IPO',\n",
    "    'INMEX',\n",
    "    'MetaboID',\n",
    "    'MRMkit',\n",
    "    'RINGS',\n",
    "    'Sieve',\n",
    "    'Lilikoi',\n",
    "    'IntLIM',\n",
    "    'RaMP',\n",
    "    'MADMAX',\n",
    "    'JUMPm',\n",
    "    'RTExtract',\n",
    "    'P-BOSS',\n",
    "    'PAPi',\n",
    "    'CTPIC'\n",
    "    \n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5fb20057-e81f-4ba8-bff3-76b981d52e6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# From Query 1\n",
    "toolsFound = set([\n",
    "    'XCMS',\n",
    "    'MetaboAnalyst',\n",
    "    'MixProTool',\n",
    "    'MaxQuant',\n",
    "    'Proteome Discoverer',\n",
    "    'Mascot',\n",
    "    'BRAIN',\n",
    "    'DiffCorr',\n",
    "    'MetMatch',\n",
    "    'metaXCMS',\n",
    "    'VOCCluster',\n",
    "    'TFsearch',\n",
    "    'MeltDB',\n",
    "    'MetDAT',\n",
    "    'Chemical Translation Service',\n",
    "    'Metandem',\n",
    "    'Pathview',\n",
    "    'Pathview Web',\n",
    "    'Sesame',\n",
    "    'MAVEN',\n",
    "    'iReMet-Flux',\n",
    "    'MetATT',\n",
    "    'MetNet3D',\n",
    "    'KegArray',\n",
    "    'ROMANCE',\n",
    "    'geoRge',\n",
    "    'X(13)CMS',\n",
    "    'MetaboLights',\n",
    "    'MetFusion',\n",
    "    'MassBank',\n",
    "    'NIST',\n",
    "    'MetFrag',\n",
    "    'XCMS Online',\n",
    "    'ConsensusPathDB',\n",
    "    'PROFANCY',\n",
    "    'AMDIS',\n",
    "    'ADAP-GC',\n",
    "    'MS-DIAL',\n",
    "    'eRah',\n",
    "    'OmicsVis',\n",
    "    'MetNet',\n",
    "    'MetCirc',\n",
    "    'MDAS',\n",
    "    'Workflow4Metabolomics',\n",
    "    'W4M',\n",
    "    'Heatmapper',\n",
    "    'OpenMS',\n",
    "    'KNIME',\n",
    "    'DRAM',\n",
    "    'DRAM-v',\n",
    "    'MZmine',\n",
    "    'MetaFIND',\n",
    "    'LiPydomics',\n",
    "    'METLIN',\n",
    "    'mzML2ISA',\n",
    "    'nmrML2ISA',\n",
    "    'INVEX',\n",
    "    'T3DB',\n",
    "    'HMDB',\n",
    "    'DrugBank',\n",
    "    'openBIS',\n",
    "    'proFIA',\n",
    "    'MetFlow',\n",
    "    'DExSI',\n",
    "    'shinyheatmap',\n",
    "    'MetaboHunter',\n",
    "    'DecoMetDIA',\n",
    "    'MetDNA',\n",
    "    'Sirius',\n",
    "    'MS-DIAL',\n",
    "    'GAM',\n",
    "    'SYNERGxDB',\n",
    "    'COMBSecretomics',\n",
    "    'MetAlign',\n",
    "    'SpectConnect',\n",
    "    'Escher',\n",
    "    'LipidXte',\n",
    "    'CPVA',\n",
    "    'SIMCA',\n",
    "    'OSI',\n",
    "    'SMMS',\n",
    "    'SMfinder',\n",
    "    'COCONETS',\n",
    "    'MetGem',\n",
    "    'BioSpider',\n",
    "    'Plant Reactome',\n",
    "    'CoPE-FBA',\n",
    "    'PARADISe',\n",
    "    'NIST',\n",
    "    'BIOMEX',\n",
    "    'PlantMetabolomics',\n",
    "    'LipidFinder',\n",
    "    'LIPID MAPS',\n",
    "    'CircadiOmics',\n",
    "    'BIO_CYCLE',\n",
    "    'CompExtractor',\n",
    "    'AMDIS-SpectConnect',\n",
    "    'MORA',\n",
    "    'KEGG',\n",
    "    'BRITE',\n",
    "    'STATegra',\n",
    "    'IIS',\n",
    "    'MRMAssayDB',\n",
    "    'SRMAtlas',\n",
    "    'PanoramaWeb',\n",
    "    'PeptideTracker',\n",
    "    'UniProtKB',\n",
    "    'MPA-RF',\n",
    "    'DIA-Umpire',\n",
    "    'MicroScope',\n",
    "    'GOcats',\n",
    "    'OWLTools',\n",
    "    'Map2Slim',\n",
    "    'PSPP',\n",
    "    'SysteMHC Atlas',\n",
    "    'DrugBank',\n",
    "    'NP-StructurePredictor',\n",
    "    'tO2PLS',\n",
    "    'SimIndex',\n",
    "    'SimZyme',\n",
    "    'PlantMAT',\n",
    "    'MassTRIX',\n",
    "    'Guineu',\n",
    "    'MassyTools',\n",
    "    'flexAnalysis'\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ddd1a679-edfa-499f-a4e9-29214e1cc7b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add wlads tools to set\n",
    "toolsFound.update(wlad_tools)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "083a5ea1-90cb-4949-bef9-6208ee0bc53a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "361"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(toolsFound)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ff0185bf-3720-404d-95f0-267a2ba6fdf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "tools = pd.Series(data = list(toolsFound), name='CuratedTools')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b7fb271b-a217-432b-9713-0512e9cdef5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "tools.to_csv('Data/CuratedTools.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
